### PR TITLE
[DOC] Service discovery

### PR DESCRIPTION
**Pull request by PaulRMellor**
Signed-off-by: prmellor <pmellor@redhat.com>

Documentation

New section to describe service discovery for:

* Internal Kafka bootstrap service
* HTTP Bridge service

May need more detail.
May not need both examples

I've added it to the _Managing Strimzi_ section, to sit with _Checking the status of a custom resource_
We can move if there's a better spot.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

**Comment by PaulRMellor** [comment]
> The text looks good. But can you clarify where will this end up? In which part of which book?

I've added it to the _Managing Strimzi_ section, to sit with _Checking the status of a custom resource_
We can move if there's a better spot.

File: documentation/modules/managing/con-service-discovery.adoc [hunk 1]
```diff
@@ -0,0 +1,83 @@
+// Module included in the following assemblies:
+//
+// managing/assembly-management-tasks.adoc
+
+[id='proc-add-service-discovery-{context}']
+= Discovering services using labels and annotations
+
+Service discovery makes it easier for client applications running in the same cluster as {ProductName} to interact with a Kafka cluster.
```
Comment (laidan6000) [inline documentation/modules/managing/con-service-discovery.adoc:8]: "the same cluster" - which type of cluster?

File: documentation/modules/managing/con-service-discovery.adoc [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+A _service discovery_ label and annotation is generated for services used to access the Kafka cluster:
+
+* Internal Kafka bootstrap service
+* HTTP Bridge service
+
+The label helps find the service, and the annotation provides connection details that a client application can use to make the connection.
```
Comment (laidan6000) [inline documentation/modules/managing/con-service-discovery.adoc:15]: "The label helps find..." is a little bit abstract. How about "makes the service discoverable", "makes the service findable", or similar wording.

File: documentation/modules/managing/con-service-discovery.adoc (hunk 1 above)
Comment (PaulRMellor) [reply documentation/modules/managing/con-service-discovery.adoc:8]: _Kubernetes cluster_

File: documentation/modules/managing/con-service-discovery.adoc (hunk 2 above)
Comment (PaulRMellor) [reply documentation/modules/managing/con-service-discovery.adoc:15]: I guess the service can be discovered without the label, but the label is making it easier. 
How about: _The label helps to make the service discoverable,_

Comment (laidan6000) [reply documentation/modules/managing/con-service-discovery.adoc]: Yes, that works for me.

**Pull request merged**